controllers/datamovementmanager_controller.go

```diff
@@ -0,0 +1,479 @@
+func (r *DataMovementManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	manager := &dmv1alpha1.DataMovementManager{}
+	if err := r.Get(ctx, req.NamespacedName, manager); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	errorHandler := func(err error, msg string) (ctrl.Result, error) {
+		if errors.IsConflict(err) {
+			return ctrl.Result{Requeue: true}, nil
+		}
+		log.Error(err, msg+" failed")
+		return ctrl.Result{}, err
+	}
+
+	if err := r.createSecretIfNecessary(ctx, manager); err != nil {
+		return errorHandler(err, "Create Secret")
+	}
+
+	if err := r.createDeploymentIfNecessary(ctx, manager); err != nil {
+		return errorHandler(err, "Create Deployment")
+	}
+
+	if err := r.createOrUpdateServiceIfNecessary(ctx, manager); err != nil {
+		return errorHandler(err, "Create Or Update Service")
+	}
+
+	if err := r.createOrUpdateDaemonSetIfNecessary(ctx, manager); err != nil {
+		return errorHandler(err, "Create Or Update DaemonSet")
+	}
+
+	manager.Status.Ready = true
+	if err := r.Status().Update(ctx, manager); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DataMovementManagerReconciler) createSecretIfNecessary(ctx context.Context, manager *dmv1alpha1.DataMovementManager) (err error) {
+	log := log.FromContext(ctx)
+
+	newSecret := func() (*corev1.Secret, error) {
+		privateKey, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
+		if err != nil {
+			return nil, fmt.Errorf("Generating private key failed: %w", err)
+		}
+		privateDER, err := x509.MarshalECPrivateKey(privateKey)
+		if err != nil {
+			return nil, fmt.Errorf("Converting private key to DER format failed: %w", err)
+		}
+
+		privatePEM := pem.EncodeToMemory(&pem.Block{
+			Type:  keyutil.ECPrivateKeyBlockType,
+			Bytes: privateDER,
+		})
+
+		publicKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
+		if err != nil {
+			return nil, fmt.Errorf("Generating public key failed: %w", err)
+		}
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      manager.Name + secretSuffix,
+				Namespace: manager.Namespace,
+			},
+			Type: corev1.SecretTypeSSHAuth,
+			Data: map[string][]byte{
+				corev1.SSHAuthPrivateKey: privatePEM,
+				sshPublicKey:             ssh.MarshalAuthorizedKey(publicKey),
+			},
+		}
+
+		if err := ctrl.SetControllerReference(manager, secret, r.Scheme); err != nil {
+			return nil, fmt.Errorf("Setting Secret controller reference failed: %w", err)
+		}
+
+		return secret, nil
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      manager.Name + secretSuffix,
+			Namespace: manager.Namespace,
+		},
+	}
+
+	if err = r.Get(ctx, client.ObjectKeyFromObject(secret), secret); errors.IsNotFound(err) {
+		secret, err = newSecret()
+		if err != nil {
+			return err
+		}
+
+		if err = r.Create(ctx, secret); err != nil {
```

Comment:
Per our slack conversation but repeated here for others  - CreateOrUpdate() compares _new_ to _old_ and since I'm always generating a new key Update() would happen every time. 
- I could use a known random seed so the private/public keys would always be the same. 
- I could also only modify the private/public keys if either is empty - but I'm not sure why that would happen. And recovery in this case would be to just delete the Secret.